src/domains.cpp

```diff
@@ -217,12 +226,26 @@ unsigned int check_local_entropy (dolfin::Function *cation,
       anion_valency,
       &adapt_voltage,
       target_mesh,
-      entropy_tol
+      entropy_tol,
+      Max_Numb_Cells
     );
     return 1+num_refines;
   }
 }
 
+unsigned int check_local_entropy (dolfin::Function *cation,
+                                  double cation_valency,
+                                  dolfin::Function *anion,
+                                  double anion_valency,
+                                  dolfin::Function *voltage,
+                                  dolfin::Mesh *target_mesh,
+                                  double entropy_tol)
+{
+  int max_numb_cells = 5E5;
```

Comment:
how was this number chosen?  If we do not pass the ``max_numb_cells`` property, i don't think we should impose the safeguard...